framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -891,6 +1029,34 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
+        if (range.from > range.to || range.from == Long.MIN_VALUE && range.to == Long.MAX_VALUE) {
+            return UNKNOWNVAL;
+        } else if (!range.isWiderThan(MAX_VALUES)) {
+            List<Long> values = new ArrayList<>();
+            for (long value = range.from; value <= range.to; value++) {
+                values.add(value);
+            }
+            return createIntValAnnotation(values);
+        }
+        AnnotationBuilder builder = new AnnotationBuilder(processingEnv, IntRange.class);
+        builder.setValue("from", range.from);
+        builder.setValue("to", range.to);
+        return builder.build();
+    }
+
+    /**
+     * For Annotation that are IntRange get the integer range
```

Comment:
What happens if the parameter is not IntRange? A runtime exception, which is ok, but should be documented.